clients/vscode-hlasmplugin/src/hlasmLanguageDetection.ts

```diff
@@ -38,14 +38,22 @@ export class HLASMLanguageDetection {
     //automatic detection function
     setHlasmLanguage(document: vscode.TextDocument): boolean {
         // check only plain text files
-        if (document.languageId == 'plaintext') {
+        if (document.languageId == 'plaintext' && !this.EndsWithExtension(document)) {
             if (this.checkHlasmLanguage(document)) {
                 vscode.languages.setTextDocumentLanguage(document, 'hlasm');
                 return true;
             }
         }
         return document.languageId == 'hlasm';
     }
+    //Checks for extension for plaintext files
+    EndsWithExtension(document: vscode.TextDocument): boolean {
+
+        var regexExtension = new RegExp("\\.\\w+$");
```

Comment:
maybe https://nodejs.org/api/path.html#path_path_extname_path ?

Reply:
Cobol Explorer has naming conventions like dataset(member). extension .
Ex :ZIMLU02.NEW.COBOL(PARTSUPP).cbl. This api will return true even for files without extension.

Reply:
ok, got it (I think), but we should probably still handle cases like '.hidden' as not having extension.